pkg/manifests/external_dns.go

```diff
@@ -54,41 +87,121 @@ func (p Provider) String() string {
 	}
 }
 
-func (p Provider) ResourceName() string {
-	switch p {
-	case PublicProvider:
-		return externalDnsResourceName
-	case PrivateProvider:
-		return externalDnsResourceName + "-private"
+// ExternalDnsConfig contains externaldns resources based on input configuration
+type ExternalDnsConfig struct {
+	//internally exposed
+	tenantId, subscription, resourceGroup,
+	clientId, serviceAccountName, namespace,
+	resourceName string
+	identityType  IdentityType
+	resourceTypes []ResourceType
+	provider      provider
+
+	// externally exposed
+	resources          []client.Object
+	labels             map[string]string
+	dnsZoneResourceIDs []string
+}
+
+func (e *ExternalDnsConfig) Resources() []client.Object {
+	return e.resources
+}
+
+func (e *ExternalDnsConfig) Labels() map[string]string {
+	return e.labels
+}
+
+func (e *ExternalDnsConfig) DnsZoneResourceIds() []string {
+	return e.dnsZoneResourceIDs
+}
+
+func NewExternalDNSConfig(conf *config.Config, tenantId, subscription, resourceGroup, clientId, inputServiceAccount, namespace, crdName string, identityType IdentityType, resourceTypes []ResourceType, provider provider, dnszoneresourceids []string) (*ExternalDnsConfig, error) {
+	// valid values for enums
+	if identityType != IdentityTypeMSI && identityType != IdentityTypeWorkloadIdentity {
+		return nil, fmt.Errorf("invalid identity type: %v", identityType)
+	}
+
+	containsGateway := false
```

Comment:
remove any crd logic. add that when we add crds

Reply:
doesn't make sense to make preemtive changes. crd changes might look different that what we want. not worth early optimizing

Reply:
done